### PR TITLE
Handle PHP Notice when get_post_type_object() returns null in is_post_type_archive()

### PR DESCRIPTION
**Pull request by srtfisher**
Trac ticket: https://core.trac.wordpress.org/ticket/56287

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (costdev) [inline .vscode/settings.json]: This file should be removed from the PR.

File: tests/phpunit/tests/query.php [hunk 2]
```diff
@@ -769,4 +769,21 @@ static function() {
 		$this->assertArrayHasKey( 'join', $posts_clauses_request );
 		$this->assertSame( '/* posts_join_request */', $posts_clauses_request['join'] );
 	}
+
+	/**
+	 * Test that is_post_type_archive() returns false for an undefined post type.
+	 *
+	 * @ticket 56287
```
Comment (costdev) [inline tests/phpunit/tests/query.php]:
```suggestion
	 * Tests that is_post_type_archive() returns false for an undefined post type.
	 *
	 * @ticket 56287
	 *
	 * @covers ::is_post_type_archive
```

File: tests/phpunit/tests/query.php [hunk 3]
```diff
@@ -769,4 +769,21 @@ static function() {
...
+	 */
+	public function test_is_post_type_archive_handles_null_post_type() {
```
Comment (costdev) [inline tests/phpunit/tests/query.php]:
```suggestion
	public function test_is_post_type_archive_should_return_false_for_an_undefined_post_type() {
```
Describing _how_ `is_post_type_archive()` should handle this case in the test name makes it easier to see what's being tested. Replacing `null_post_type` with `an_undefined_post_type` is clearer and also matches the test docblock.

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the PR and unit tests! I've left some small change requests in this review.

**Comment by srtfisher** [comment]
@costdev Thanks for the review! Changes made.

**Review by costdev — APPROVED** [review]
PR looks good to me provided manual testing works as expected. :+1:

**Review by audrasjb — APPROVED** [review]
Looks good.

**Comment by audrasjb** [comment]
Committed in https://core.trac.wordpress.org/changeset/54464

**Pull request closed without merging**